### ISSUE TITLE
PowerCube addition failed - V5FrameError: V5 Modbus EXCEPTION: AcknowledgeError

### ISSUE THREAD

**Issue by AndyNew2** [davidrapan/ha-solarman#329]
### Description

I tried to configure a Pylontech Batterie which is equipped with a Solarman WLAN stick.
I found in the log following error message:
2024-12-17 21:57:42.003 ERROR (MainThread) [homeassistant.config_entries] Config entry 'Pylontech' from integration solarman has an invalid unique_id '400nnnnnnn' of type int when a string is expected, please create a bug report at https://github.com/davidrapan/ha-solarman/issues
2024-12-17 21:57:42.003 ERROR (MainThread) [homeassistant.config_entries] Config entry 'Pylontech' from integration solarman has an invalid unique_id '400nnnnnnn' of type int when a string is expected, please create a bug report at https://github.com/davidrapan/ha-solarman/issues

It seems there is a logic error, because HA expects a string and do not like the serial number, which consist only of numbers. So nnnnnnn had been replaced with my real serial number (don't want to post my full serial number)

At the GUI I see this:
V5 Modbus EXCEPTION: AcknowledgeError

Seems something went completely wrong with this. I have a Sofar Inverter already running and working. This was just the try to add the batterie information to it...

### Reproduction Steps

configure Pylontech H2 Force Batterie with regular steps. Not sure if it hurts, because I have already an inverter connected and working. So this would be the second instance of solarman....

### Log

2024-12-17 21:57:42.003 ERROR (MainThread) [homeassistant.config_entries] Config entry 'Pylontech' from integration solarman has an invalid unique_id '400......0' of type int when a string is expected, please create a bug report at https://github.com/davidrapan/ha-solarman/issues
2024-12-17 21:57:42.003 ERROR (MainThread) [homeassistant.config_entries] Config entry 'Pylontech' from integration solarman has an invalid unique_id '400......0' of type int when a string is expected, please create a bug report at https://github.com/davidrapan/ha-solarman/issues
2

### Version

24.12.16

### Home Assistant Version

2024.12.4

**Comment by davidrapan** [comment]
Please try to add device w/ debug log enabled and attach the file.

**Comment by AndyNew2** [comment]
Here attached. I replaced the serial numbers (keep first and last digits and replaced the middle part with n, but had been all numbers). Hope that information is no further relevant. You see there are two devices, one works the other starting with 40 fails...
[home-assistant_solarman_2024-12-17T21-22-47.195Z.log](https://github.com/user-attachments/files/18171608/home-assistant_solarman_2024-12-17T21-22-47.195Z.log)

**Comment by davidrapan** [comment]
So it's a problem w/ replies from your device as it's returning a modbus exception 0x05.

I would try to strip out the original file of some registers. Particularly those in the 03 ~ 4096 - 4106 | 0x1000 - 0x100A range and try it w/o them.

But I won't be much help here as I have zero experience w/ this type of a device... :-/

**Comment by AndyNew2** [comment]
You are right. I tried your recommendation to strip out suspisious registers. No change.
I have given up to integrate the Pylontech Solarman integration into HA. By the way, even in Solarman webpage there is no useful data. My crazy PowerCube has a useless Solarman WLAN Stick, for nothing good, just to consume a little bit more engery...

Close this topic.

**Comment by davidrapan** [comment]
Uh, I'm sorry to hear that! You could try to contact PowerCube manufacturer and see if they are willing to provide modbus rtu documentation. I think it's worth the shot. 😉
